src/service/userDetails.js

```diff
@@ -159,9 +164,33 @@ class UserDetails {
     }
   }
 
+  static approve(req, res) {
+    try {
+      // TODO: How are we ensuring that the currently logged-in user is privileged to perform this action?
+      return userDetails
+        .findByPk(req.params.id)
+        .then((details) => {
+          details
+            .update({
+              approved: true,
+              updatedAt: new Date(),
+            })
+            .then((_updateduserDetails) => {
+              res.status(200).send({
+                message: 'user approved successfully',
+              });
+            })
+            .catch(error => res.status(400).send(error));
+        })
+        .catch(error => res.status(400).send(error));
+    } catch (e) {
+      res.status(500).send({ error: e.message });
+    }
+  }
+
   static login(req, res) {
     try {
-      userDetails.findOne({ where: { loginId: req.body.loginId } }).then((user) => {
+      userDetails.findOne({ where: { loginId: req.body.loginId, approved: true } }).then((user) => {
```

Comment:
I hope we can find the user just by login Id and then decide the code flow based on persona and approved status.

For now the field "approved" describes the buyer's status. So there is a chance that this endpoint might block admin's login if we don't do a data migration setting approved true for admin.

Please correct me, if there is any gape in my understanding.

Reply:
good catch @nagarajankannan - will make the modification.

Reply:
Done - please review once again.